### ISSUE TITLE
Construction Mode: Switch hog can be bought even if it is already infinitely times available

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmith/hedgewars#859]
```
In Construction Mode, you can buy/generate/whatever a switch hog utilty from 
the support station, even if the weapon scheme already gives you the item 
infinitely times.

This is not really a bug, but slightly annoying, as it just clutters the menu.

Since Construction Mode allows custom schemes, I guess the general solution to 
this would be to remove everything which is already available infinitely times 
from the set of “buyable” items.


I tested Construction mode in Mercurial r5ae7ba0b2849 (aka beta for 0.9.21) on 
GNU/Linux.
```

Original issue reported on code.google.com by `almikes@aol.com` on 11 Dec 2014 at 9:02

**Comment by GoogleCodeExporter** [comment]
Original comment by `sheepyluva` on 13 Dec 2014 at 9:30

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the report / testing.
Fixed in 2f19ff0ded73.
```

Original comment by `RedGrin...@gmail.com` on 15 Dec 2014 at 3:28
* Changed state: **Fixed**
